### PR TITLE
Installation guide

### PR DESCRIPTION
**Pull request by kishorj**
Documentation for the following:
- Controller Installation guide
- Configuration options
- Subnet auto-discovery

### CODE REVIEW

File: docs/guide/controller/installation.md [hunk 1]
```diff
@@ -1 +1,70 @@
...
+## Via Helm
+Follow the instructions in [aws-load-balancer-controller](https://github.com/aws/eks-charts/tree/master/stable/aws-load-balancer-controller) helm chart.
+
+## Via Yaml manifests
+
+### Migrating from AWS ALB Ingress controller
+If AWS ALB Ingress controller is installed, refer to [migrating from v1 to v2](../upgrade/migrate_v1_v2.md)
+
+!!!warning ""
+    AWS ALB inress controller must be uninstalled before installing AWS Load Balancer controller.
```
Comment (M00nF1sh) [inline docs/guide/controller/installation.md:12]: inress  -> Ingress

File: docs/guide/controller/installation.md [hunk 2]
```diff
@@ -1 +1,70 @@
...
+#### Upgrading from ALB ingress controller
+If migrating from ALB ingress controller, grant [additional IAM permissions](../../install/iam_policy_v1_to_v2_additional.json).
+
+### Install cert-manager
+- For Kubernetes 1.16+: `kubectl apply --validate=false -f https://github.com/jetstack/cert-manager/releases/download/v1.0.2/cert-manager.yaml`
+- For Kubernetes <1.16: `kubectl apply --validate=false -f https://github.com/jetstack/cert-manager/releases/download/v1.0.2/cert-manager-legacy.yaml`
+
+### Download and apply the yaml spec
+- curl -o https://raw.githubusercontent.com/kubernetes-sigs/aws-alb-ingress-controller/v2_ga/config/samples/install_v2_0_0.yaml
+- Edit the saved yaml file, go to line 452 and set the --cluster-name arg value to your EKS cluster name
```
Comment (M00nF1sh) [inline docs/guide/controller/installation.md:65]: we don't need to mention the line number as it's subject to change

File: docs/guide/controller/installation.md (hunk 1 above)
Comment (kishorj) [reply docs/guide/controller/installation.md:12]: ack

File: docs/guide/controller/installation.md (hunk 2 above)
Comment (kishorj) [reply docs/guide/controller/installation.md:65]: ok, I will put generic references.

**Pull request merged**